spec/onlinerParser.ts

```diff
@@ -0,0 +1,150 @@
+            model: {
+              id: '2',
+              name: 'X3'
+            },
+            odometerState: 52200,
+            year: 2010
+          },
+          id: 2,
+          photos: [] as any[]
+        }
+      },
+      content: `
+        <tr class="carRow">
+          <td class="txt">
+            <p>внедорожник, бензин 3.7 л</p>
+          </td>
+          <td>
+            <div class="cost-i">
+              <p class="small">7000 $<br> 6018 €<br></p>
+            </div>
+          </td>
+        </tr>
+        <tr class="carRow">
+          <td class="txt">
+            <p>внедорожник, бензин 3.7 л</p>
+          </td>
+          <td>
+            <div class="cost-i">
+              <p class="small">8000 $<br> 7018 €<br></p>
+            </div>
+          </td>
+        </tr>
+      `
+    },
+    ADS_RESULT: {
+      '1': {
+        car: { body: 5, model: { id: '1', name: 'X5 M' }, odometerState: 198000, year: 2011 },
+        description: 'внедорожник, бензин 3.7 л',
+        id: 1,
+        photos: [] as any[],
+        price: '7000'
+      },
+      '2': {
+        car: { body: 5, model: { id: '2', name: 'X3' }, odometerState: 52200, year: 2010 },
+        description: 'внедорожник, бензин 3.7 л',
+        id: 2,
+        photos: [] as any[],
+        price: '8000'
+      }
+    },
+    BODY_TYPES: `
+      <li class="body_type-1">
+        <label>
+          <input type="checkbox" name="body_type[]" class="f-cb" value="1" />
+          Седан&nbsp;<small class="count"></small>
+        </label>
+      </li>
+      <li class="body_type-2">
+        <label>
+          <input type="checkbox" name="body_type[]" class="f-cb" value="2" />
+          Универсал&nbsp;<small class="count"></small>
+        </label>
+      </li>
+    `,
+    BODY_TYPES_RESULT: ['Седан', 'Универсал'],
+    MARKS: `var Manufactures = [{"id":1,"name":"Acura"},{"id":139,"name":"Adler"}];`,
+    MARKS_RESULT: [{ id: 1, name: 'Acura' }, { id: 139, name: 'Adler' }],
+    MODELS: `var ManufacturesModel = {"183":[{"'33 Hot Rod":"2581"},{"818":"2583"}]};`,
+    MODELS_RESULT: { '183': [{ "'33 Hot Rod": '2581' }, { '818': '2583' }] }
+  };
+
+  it('should be return an array of marks', async () => {
+    const puppeteerStub = sinon.stub(puppeteer, 'launch').returns({
+      close: () => '',
+      newPage: () => ({
+        content: () => parser.MARKS,
+        goto: (url: string, options: any) => ''
+      })
+    });
+    await api.updateMarks();
+    assert.deepEqual(api.getMarks() as any, parser.MARKS_RESULT);
+    puppeteerStub.restore();
+  });
+
+  it('should be return an array of models', async () => {
+    const puppeteerStub = sinon.stub(puppeteer, 'launch').returns({
+      close: () => '',
+      newPage: () => ({
+        content: () => parser.MODELS,
+        goto: (url: string, options: any) => ''
+      })
+    });
+    await api.updateModels();
+    assert.deepEqual(api.getModels() as any, parser.MODELS_RESULT);
+    puppeteerStub.restore();
```

Comment:
All restore calls should happen only inside the finally {} block.
If this is not true in case of any error the method will be stabbed for the rest of the tests.